npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -380,9 +380,10 @@ def clean_npmrc_in_path(yarn_lock)
           dirs.pop
           while dirs.any?
             npmrc = dirs.join("/") + "/.npmrc"
-            break unless File.exist?(npmrc)
-
-            File.write(npmrc, File.read(npmrc).gsub(/\$\{.*\}/, ""))
+            if File.exist?(npmrc)
+              # If the .npmrc file exists, clean it
+              File.write(npmrc, File.read(npmrc).gsub(/\$\{.*?\}/, ""))
```

Comment:
The change to make this non-greedy makes sense to me. I think something like `\$\{[^}]*\}` would have the same effect, correct?

Is this is just something that you observed would be more robust when working on this, or part of this bug fix?

Reply:
@deivid-rodriguez: I was getting the suggestion from the copilot for the `non-greedy matching` while typing the `File.write`, so did little dig on it and thought of adding it to this method.

Reply:
Aaah makes sense. Thanks!